src/common/s3_adapter.cpp

```diff
@@ -478,7 +489,12 @@ void S3Adapter::GetObjectAsync(std::shared_ptr<GetObjectAsyncContext> context) {
         throttle_->Add(true, context->len);
     }
 
-    inflightBytesThrottle_->OnStart(context->len);
+    if (inflightBytesThrottle_->OnStart(context->len) < 0) {
+        context->retCode = -1;
+        context->retryIntervalUS = retryIntervalUS_;
+        asyncCallbackTP_.Enqueue(context->cb, this, context);
```

Comment:
too many throttled requests will make callback queue too large?

Reply:
put/get 10 objects async requests from the same function, a for loop

before: 1st success, 2nd throttled (on start cond_.wait), 3\~10 blocked
now: 1st success, 2\~10 throttled but still put callbacks to queue


if for loop is too large, such as download all warmup objs (>100,000 maybe), queue size maybe too large? how about the memory consumption? need more test data.